java-checks/src/main/java/org/sonar/java/checks/RawByteBitwiseOperationsCheck.java

```diff
@@ -0,0 +1,107 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.java.checks.helpers.ExpressionsHelper;
+import org.sonar.java.tag.Tag;
+import org.sonar.plugins.java.api.JavaFileScanner;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.semantic.Type;
+import org.sonar.plugins.java.api.semantic.Type.Primitives;
+import org.sonar.plugins.java.api.tree.BaseTreeVisitor;
+import org.sonar.plugins.java.api.tree.BinaryExpressionTree;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.squidbridge.annotations.ActivatedByDefault;
+import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
+import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
+
+import java.util.LinkedList;
+import java.util.List;
+
+@Rule(
+  key = "S3034",
+  name = "Raw byte values should not be used in bitwise operations in combination with shifts",
+  priority = Priority.CRITICAL,
+  tags = {Tag.BUG})
+@ActivatedByDefault
+@SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.INSTRUCTION_RELIABILITY)
+@SqaleConstantRemediation("5min")
+public class RawByteBitwiseOperationsCheck extends BaseTreeVisitor implements JavaFileScanner {
+
+  JavaFileScannerContext context;
+  List<ExpressionTree> shifts = new LinkedList<>();
+  List<ExpressionTree> byteSecuring = new LinkedList<>();
+
+
+  @Override
+  public void scanFile(JavaFileScannerContext context) {
+    this.context = context;
+    scan(context.getTree());
+    shifts.clear();
+    byteSecuring.clear();
+  }
+
+  @Override
+  public void visitBinaryExpression(BinaryExpressionTree tree) {
+    super.visitBinaryExpression(tree);
+    if (isShifting(tree)) {
+      shifts.add(tree);
+      return;
+    }
+    if (isSecuringByte(tree)) {
+      byteSecuring.add(tree);
+      return;
+    }
+    if (isIntegerOrLongExpected(tree.symbolType())) {
+      ExpressionTree leftOperand = ExpressionsHelper.skipParentheses(tree.leftOperand());
+      ExpressionTree rightOperand = ExpressionsHelper.skipParentheses(tree.rightOperand());
+      checkShiftWithoutByteSecuring(leftOperand, rightOperand);
+      checkShiftWithoutByteSecuring(rightOperand, leftOperand);
+    }
+  }
+
+  private static boolean isShifting(BinaryExpressionTree tree) {
+    return tree.is(Tree.Kind.LEFT_SHIFT, Tree.Kind.RIGHT_SHIFT, Tree.Kind.UNSIGNED_RIGHT_SHIFT);
```

Comment:
Yes, because the issue is not with the shift itself, but with the addition of a signed byte , not protected, afterward.